martin/src/pg/config.rs

```diff
@@ -110,10 +128,12 @@ impl PgConfig {
             self.auto_publish = OptBoolObj::Bool(true);
         }
 
+        self.validate()?;
         Ok(res)
     }
 
     pub async fn resolve(&mut self, id_resolver: IdResolver) -> MartinResult<TileInfoSources> {
+        self.validate()?;
```

Comment:
why do we need to validate it in so many places - here, in the pool, and in finalize?  I don't remember for sure which codepath always executes, but we should probably limit it to just one. Probably finalize? Please check. Thx!